cadc-vos-server/src/main/java/org/opencadc/vospace/server/transfers/PushToVOSpaceNegotiation.java

```diff
@@ -112,30 +111,16 @@ public void doAction()
 
             VOSURI target = new VOSURI(transfer.getTargets().get(0));
 
-            PathResolver resolver = new PathResolver(nodePersistence, authorizer, true);
+            PathResolver pr = new PathResolver(nodePersistence, authorizer, true);
+            PathResolver.Resolved resolved = pr.resolveNode(target);
+            log.debug("Resolved target node: " + resolved);
 
-            Node n = resolver.getNode(target.getParentURI().getPath());
-            if (!(n instanceof ContainerNode)) {
-                throw new TransferException("parent is not a container node");
-            }
-            ContainerNode parent = (ContainerNode) n;
-            log.debug("found parent: " + parent);
+            DataNode dn;
             Subject caller = AuthenticationUtil.getCurrentSubject();
-            if (!authorizer.hasSingleNodeWritePermission(parent, caller)) {
-                throw NodeFault.PermissionDenied.getStatus(target.getParentURI().getURI().toASCIIString());
-            }
-            
-            //Node node = resolveNodeForWrite(authorizer, nodePersistence, target, DataNode.class, true, true, true);
-            Node node = nodePersistence.get(parent, target.getName());
-            log.debug("target node: " + target + " -> " + node);
-            
-            DataNode dn = null;
-            if (node == null) {
-                if (!authorizer.hasSingleNodeWritePermission(parent, caller)) {
-                    throw NodeFault.PermissionDenied.getStatus(target.getParentURI().getURI().toASCIIString());
-                }
+            if (resolved.child == null) {
```

Comment:
this is now missing two checks:
`resolved.parent != null` 
`hasSingleNodeWritePermission(resolved.parent, caller)`

Reply:
It's `resolveNode` responsibility to return a valid `parent` and `childName` when `child` not found. Good point about `hasSingleNodeWritePermission(resolved.parent, caller)`. Fixed